src/client/client.rs

```diff
@@ -465,6 +465,29 @@ impl Client {
         Ok(())
     }
 
+    /// Split a Client into a read half and a publish half, which can be used to read and publish concurrently.
+    pub fn into_split(mut self) -> Result<(ClientReadHalf, ClientPublishHalf)> {
```

Comment:
I think it would probably make sense to add an example how to use this to the documentation. Besides making it easier to use I guess it can also serve as a doc test. Although probably should be marked as `no_run`.